service/deferrer/service.go

```diff
@@ -31,3 +43,90 @@ func New(config Config) (*Service, error) {
 
 	return s, nil
 }
+
+// ShouldDefer finds corresponding DrainerConfig for the POD it's running in
+// and checks if node in current POD is drained yet. If DrainerConfig doesn't
+// exist or doesn't have Drained or Timeout condition, node termination should
+// be deferred.
+//
+// Current POD name and namespace are picked from environment variables with
+// corresponding keys defined in constants EnvKeyMyPodName &
+// EnvKeyMyPodNamespace. Defining these env variables is most conveniently
+// achieved by utilizing Kubernetes Downward API:
+// https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/
+func (s *Service) ShouldDefer(ctx context.Context) (bool, error) {
+	var err error
+
+	var podName, podNamespace string
+	{
+		s.logger.LogCtx(ctx, "level", "debug", "message", "finding pod name and namespace")
+
+		podName, err = s.getPodName()
+		if err != nil {
+			return false, microerror.Mask(err)
+		}
+		podNamespace, err = s.getPodNamespace()
+		if err != nil {
+			return false, microerror.Mask(err)
+		}
+
+		s.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found pod name and namespace: %s, %s", podName, podNamespace))
+	}
+
+	var drainerConfig *v1alpha1.DrainerConfig
+	{
+		s.logger.LogCtx(ctx, "level", "debug", "message", "finding drainerconfig for pod")
+
+		drainerConfig, err = s.g8sClient.CoreV1alpha1().DrainerConfigs(podNamespace).Get(podName, metasv1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			s.logger.LogCtx(ctx, "level", "debug", "message", "did not find drainerconfig")
+			drainerConfig = nil
+		} else if err != nil {
+			return true, microerror.Mask(err)
+		}
+
+		s.logger.LogCtx(ctx, "level", "debug", "message", "found drainerconfig for pod")
+	}
+
+	{
+		s.logger.LogCtx(ctx, "level", "debug", "message", "finding if node termination has to be defered")
+
+		if drainerConfig == nil {
+			s.logger.LogCtx(ctx, "level", "debug", "message", "found node termination should be deferred")
+			s.logger.LogCtx(ctx, "level", "debug", "message", "pod drainerconfig does not exist")
+			return true, nil
+		}
+
+		if drainerConfig.Status.HasDrainedCondition() {
+			s.logger.LogCtx(ctx, "level", "debug", "message", "found node termination does not have to be deferred")
+			s.logger.LogCtx(ctx, "level", "debug", "message", "pod drainerconfig has drained condition")
+			return false, nil
+		}
+		if drainerConfig.Status.HasTimeoutCondition() {
+			s.logger.LogCtx(ctx, "level", "debug", "message", "found node termination does not have to be deferred")
+			s.logger.LogCtx(ctx, "level", "debug", "message", "pod drainerconfig has timeout condition")
+			return false, nil
+		}
+
+		s.logger.LogCtx(ctx, "level", "debug", "message", "found node termination should be deferred")
```

Comment:
Note for the future. Warning: I'm nitpicky here :D 

If we have:
`finding if node termination has to be deferred`
Then we should have: 
`found node termination has to be deferred`
Or:
`found node termination does not have to be deferred`

If we have:
`finding if node termination should be deferred`
Then we should have: 
`found node termination should be deferred`
Or:
`found node termination should not be deferred`

ATM this is mixed.